scripts/Muon/GUI/Common/phase_table_widget/phase_table_presenter.py

```diff
@@ -0,0 +1,151 @@
+from Muon.GUI.Common.thread_model_wrapper import ThreadModelWrapper
+from Muon.GUI.Common import thread_model
+from Muon.GUI.Common.utilities.algorithm_utils import run_CalMuonDetectorPhases, run_PhaseQuad
+from Muon.GUI.Common.observer_pattern import Observer, Observable
+from mantid.api import AnalysisDataService
+import re
+
+
+class GenericObserver(Observer):
```

Comment:
dont we have this already in https://github.com/mantidproject/mantid/blob/master/scripts/Muon/GUI/Common/observer_pattern.py ? If not I think that would be a better home for it.